ecl2df/common.py

```diff
@@ -627,43 +628,87 @@ def stack_on_colnames(dframe, sep="@", stackcolname="DATE", inplace=True):
     return dframe
 
 
-def parse_zonemapfile(filename: str):
-    """Return a dictionary from (int) K layers in the simgrid to strings
+def _is_color(input_string: str) -> bool:
+    """Check if the input string is a 6 digit hexadecimal color code"""
+    if input_string.startswith("#") and len(input_string) == 7:
+        return True
+    return False
 
-    Typical usage is to map from grid layer to zone names.
 
-    The layer filename must currently follow format:
+def parse_lyrfile(filename: str) -> Dict[str, Dict[str, Any]]:
+    """Return a dictionary presentation of the lyr file.
 
-      'ZoneA' 1-4
-      'ZoneB' 5-10
+    The lyr file contains data of the following format,
```

Comment:
Is there an URL (resinsight.org?) that defines this format that we can refer to?

Reply:
Yes: https://resinsight.org/3d-main-window/formations/#formation-names-description-files-_lyr_

Reply:
There are still some functionality we are not handling. F.ex this. That is a bit tricky to solve.
![image](https://user-images.githubusercontent.com/47847084/117780641-1bdc3700-b240-11eb-8a98-2d04d6825476.png)

Reply:
and the SVG color keyword names.

Reply:
It's not a single layer format, it's 3 additional layers relative to the previous line. So that adds quite a lot of logic to the code. Should we also check that zones don't have overlapping layers?

Reply:
But we can't accept everything as an color. F.ex:
`ZoneA 3     -10`

This shouldn't be interpreted as 3 layers with colorcode "-10"

Reply:
good point

Reply:
I agree with your comment about making the output a list of dicts